src/snowflake/snowpark/_internal/data_source/dbms_dialects/base_dialect.py

```diff
@@ -9,6 +9,22 @@
 class BaseDialect:
     @staticmethod
     def generate_select_query(
-        table_or_query: str, schema: StructType, raw_schema: List[tuple], is_query: bool
+        table_or_query: str,
+        schema: StructType,
+        raw_schema: List[tuple],
+        is_query: bool,
+        query_input_alias: str,
     ) -> str:
-        return f"SELECT * FROM {table_or_query}"
+        cols = []
+        for raw_field in raw_schema:
+            if is_query:
+                cols.append(
+                    f"""{query_input_alias}.`{raw_field[0]}` AS {raw_field[0]}"""
+                )
+            else:
+                cols.append(raw_field[0])
+
+        if is_query:
+            return f"""SELECT {" , ".join(cols)} FROM ({table_or_query}) {query_input_alias}"""
```

Comment:
is this compatible with our existing implementation?

today we need users to give a string wrapped in `()`.  will redundant parentheses break sql?

suppose users give us `(select * from some_table)`, we will have then "select * from ((select * from some_table)) some_alias". would this work for DBMSes?

Reply:
just tested it, DBMS would ignore the redundant `()`, I added some test for this